xmtp_mls/src/groups/tests/test_libxmtp_version.rs

```diff
@@ -0,0 +1,116 @@
+use crate::groups::validated_commit::{CommitValidationError, LibXMTPVersion};
+
+#[test]
+fn test_parse_and_compare_basic_versions() {
+    let v1_0_0 = LibXMTPVersion::parse("1.0.0").unwrap();
+    let v1_0_1 = LibXMTPVersion::parse("1.0.1").unwrap();
+    let v1_1_0 = LibXMTPVersion::parse("1.1.0").unwrap();
+    let v2_0_0 = LibXMTPVersion::parse("2.0.0").unwrap();
+
+    // Test patch version comparison
+    assert!(v1_0_0 < v1_0_1);
+    assert!(v1_0_1 > v1_0_0);
+
+    // Test minor version comparison
+    assert!(v1_0_1 < v1_1_0);
+    assert!(v1_1_0 > v1_0_1);
+
+    // Test major version comparison
+    assert!(v1_1_0 < v2_0_0);
+    assert!(v2_0_0 > v1_1_0);
+
+    // Test equality
+    let v1_0_0_dup = LibXMTPVersion::parse("1.0.0").unwrap();
+    assert_eq!(v1_0_0, v1_0_0_dup);
+}
+
+#[test]
+fn test_parse_and_compare_with_suffixes() {
+    let v1_0_0 = LibXMTPVersion::parse("1.0.0").unwrap();
+    let v1_0_0_alpha = LibXMTPVersion::parse("1.0.0-alpha").unwrap();
+    let v1_0_0_beta = LibXMTPVersion::parse("1.0.0-beta").unwrap();
+    let v1_0_0_rc1 = LibXMTPVersion::parse("1.0.0-rc1").unwrap();
+    let v1_0_1_alpha = LibXMTPVersion::parse("1.0.1-alpha").unwrap();
+
+    // Versions with suffixes compare lexicographically by suffix after version parts
+    assert!(v1_0_0_alpha < v1_0_0_beta);
+    assert!(v1_0_0_beta < v1_0_0_rc1);
+
+    // Version without suffix (None) is less than version with suffix (Some)
+    // because None < Some in Rust's default Ord implementation
+    // NOTE: this does not match semver comparison
+    assert!(v1_0_0 < v1_0_0_alpha);
+    assert!(v1_0_0 < v1_0_0_beta);
```

Comment:
Given that a version without suffix is less than version with suffix, we may want this feature to be released with a version number higher than 1.6.0 (say 1.6.1), since 1.6.0-dev is already in use by some development builds that are under testing. If an app using 1.6.0-dev build without the ability to handle readd commit validation received such a commit, than they would fork. This fallout would be limited to development builds, and would not affect prod versions. But this is still worth considering.